File: nils-tdt-4100/foreksempel/src/main/java/uke2/fasit/DownCounter.java
package uke2.fasit;

public class DownCounter {
    int counter;

    public DownCounter(int counter){
        this.counter = counter;
    }

    public boolean isFinished() {
        return counter == 0;
    }

    public void countDown() {
        if (counter > 0) {
            counter -= 1;
        }    
}

    public static void main(String[] args) {
        DownCounter dc = new DownCounter(4);
        for (int i = 0; i < 5; i++) {
            System.out.println(dc.isFinished());
            dc.countDown();
        }
    }
}

File: nils-tdt-4100/foreksempel/src/main/java/uke2/DownCounter.py
def isFinished():
    return counter == 0;

def countDown():
    if not isFinished():
        global counter
        counter -= 1
        print('Teller ned med 1, nå er det',counter)
        
counter = 4
countDown()
print(isFinished())
countDown()
print(isFinished())
countDown()
print(isFinished())
countDown()
print(isFinished())
countDown()
print(isFinished())

# Når vi kjører dette scriptet er vi faktisk inni 'dette scriptet' sin main-metode!
# Hvis du vil lære mer om dette: 
# https://stackoverflow.com/questions/419163/what-does-if-name-main-do
# Dette virker litt merkelig, men vi pleier ofte å skrive slikt som dette:

if __name__ == "__main__":
    print("Hello, World!")
    # Her kan en gjerne lage kall til metodene en har laget, for å teste dem.
    # __name__ er ikke __main__ når man importerer denne koden via import.
# 